ui/src/main.rs

```diff
@@ -399,6 +405,38 @@ pub fn app_bootstrap(cx: Scope) -> Element {
     };
     state.ui.metadata = window_meta;
 
+    // load any extensions, we currently don't care to store the result of located extensions since they are stored by the librarian.
+    // We should however ensure we use the same librarian across the app so they should probably live in a globally accessible place
+    // that updates when they have new info, i.e. state.
+    fs::create_dir_all(&STATIC_ARGS.extensions_path).unwrap();
+    let paths = fs::read_dir(&STATIC_ARGS.extensions_path).expect("Directory is empty");
+    let mut extensions_library = AvailableExtensions::new();
+
+    for entry in paths {
+        let path = entry.unwrap().path();
+        println!("{:?}", path);
```

Comment:
why use `println!` instead of `log`? executing `ui` with the `debug`, `trace`, or `trace2` flag causes the log macros to send output to the terminal. you can view available options like this: `ui --help`

Reply:
debugging, pr's still not ready.